device/src/lib.rs

```diff
@@ -254,3 +254,17 @@ where
         result
     }
 }
+
+/// Trait used to mark types which can give out an exclusice reference to [`RngCore`].
+/// This trait is an implementation detail and should not be implemented outside this crate.
+#[doc(hidden)]
+pub trait GetRng: private::Sealed {
```

Comment:
pub(crate)?

Reply:
It's unfortunate, but `GetRng` has to be public, otherwise it's a compile error (can't leak private trait). That's why there's the `Sealed` trait, as a workaround to prevent other crates from providing their own implementations.